src/core/OSTreeTUI.cpp

```diff
@@ -1,292 +1,395 @@
+            SetPromotionMode(true, visibleCommitViewMap.at(selectedCommit));
+        }
+        // copy commit id
+        if (event == Event::AltC) {
+            std::string hash = visibleCommitViewMap.at(selectedCommit);
+            clip::set_text(hash);
+            notificationText = " Copied Hash " + hash + " ";
+            return true;
+        }
+        // refresh repository
+        if (event == Event::AltR) {
+            RefreshOSTreeRepository();
+            notificationText = " Refreshed Repository Data ";
+            return true;
+        }
+        // exit
+        if (event == Event::AltQ) {
+            screen.ExitLoopClosure()();
+            return true;
+        }
+        // make commit list focussable
+        if (event == Event::ArrowLeft && managerRenderer->Focused() &&
+            manager->getTabIndex() == 0) {
+            commitListComponent->TakeFocus();
+            return true;
+        }
+        return false;
     });
+}
 
-	// filter
-	BranchBoxManager filterManager(ostreeRepo, visibleBranches);
-	Component filterView = Renderer(filterManager.branchBoxes, [&] {
-		return filterManager.branchBoxRender();
-	});
-
-	// promotion
-	ContentPromotionManager promotionManager(showTooltips);
-	promotionManager.setBranchRadiobox(Radiobox(&allBranches, &promotionManager.selectedBranch));
-	promotionManager.setApplyButton(Button(" Apply ", [&] {
-		ostreeRepo.promoteCommit(visibleCommitViewMap.at(selectedCommit),
-								  ostreeRepo.getBranches().at(static_cast<size_t>(promotionManager.selectedBranch)),
-								  {}, promotionManager.newSubject,
-								  true);
-		refresh_repository();
-		notificationText = " Applied content promotion. ";
-	}, ButtonOption::Simple()));
-	Component promotionView = Renderer(promotionManager.composePromotionComponent(), [&] {
-		if (visibleCommitViewMap.size() <= 0) {
-			return text(" please select a commit to continue commit-promotion... ") | color(Color::RedLight) | bold | center;
-		}
-		return promotionManager.renderPromotionView(ostreeRepo, screen.dimy(),
-			ostreeRepo.getCommitList().at(visibleCommitViewMap.at(selectedCommit)));
+int OSTreeTUI::Run() {
+    using namespace ftxui;
+    // footer notification update loader
+    // Probably not the best solution, having an active wait and should maybe
+    // only be started, once a notification is set...
+    bool runSubThreads{true};
+    std::thread footerNotificationUpdater([&] {
+        while (runSubThreads) {
+            using namespace std::chrono_literals;
+            // notification is set
+            if (notificationText != "") {
+                footer.SetContent(notificationText);
+                screen.Post(Event::Custom);
+                std::this_thread::sleep_for(2s);
+                // clear notification
+                notificationText = "";
+                footer.ResetContent();
+                screen.Post(Event::Custom);
+            }
+            std::this_thread::sleep_for(0.2s);
```

Comment:
Does not need to be fixed now, but a [`std::condition_variable`](https://en.cppreference.com/w/cpp/thread/condition_variable) might be the better solution.